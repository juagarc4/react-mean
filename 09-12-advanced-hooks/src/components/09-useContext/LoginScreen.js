import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {
  const { setUser } = useContext(UserContext)
  const newUser = { id: 1, name: 'Raul', email: 'email@rgc.com' }
  return (
    <div>
      <h1>LoginScreen</h1>
      <hr />
      <button className="btn btn-primary" onClick={() => setUser(newUser)}>
        Login
      </button>
    </div>
  )
}
