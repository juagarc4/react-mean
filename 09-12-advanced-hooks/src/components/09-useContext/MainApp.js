import React from 'react'
import { AppRouter } from './AppRouter'
import { UserContext } from './UserContext'
export const MainApp = () => {
  const user = {
    id: 1,
    name: 'Raul Garcia',
    email: 'email@rgc.com',
  }
  return (
    <UserContext.Provider value={user}>
      <AppRouter />
    </UserContext.Provider>
  )
}
