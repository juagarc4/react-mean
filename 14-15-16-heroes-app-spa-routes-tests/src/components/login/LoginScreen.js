import React, { useContext } from 'react'
import { AuthContext } from 'auth/AuthContext'
import { types } from 'types/types'

export const LoginScreen = ({ history }) => {
  const { dispatch } = useContext(AuthContext)

  const handleLogin = () => {
    const lastVisited = localStorage.getItem('lastVisited') || '/'

    dispatch({
      type: types.login,
      payload: {
        name: 'Raul',
      },
    })
    history.replace(lastVisited)
  }
  return (
    <div className="container mt-5">
      <h1>Login Screen</h1>
      <hr />

      <button className="btn btn-primary" onClick={handleLogin}>
        Login
      </button>
    </div>
  )
}
