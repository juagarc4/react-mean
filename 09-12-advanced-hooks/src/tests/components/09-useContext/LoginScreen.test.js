import React from 'react'
import { shallow, mount } from 'enzyme'
import { UserContext } from 'components/09-useContext/UserContext'
import { LoginScreen } from 'components/09-useContext/LoginScreen'

describe('Tests in component <HomeScreen/>', () => {
  const user = {
    id: 1,
    name: 'Raul',
    email: 'email@rgc.com',
  }
  const setUser = jest.fn()
  const wrapper = mount(
    <UserContext.Provider value={{ setUser }}>
      <LoginScreen />
    </UserContext.Provider>
  )

  test('Should show the component properly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('Should execute setUser with the expected argument', () => {
    wrapper.find('button').simulate('click')
    expect(setUser).toHaveBeenCalledWith(user)
  })
})
