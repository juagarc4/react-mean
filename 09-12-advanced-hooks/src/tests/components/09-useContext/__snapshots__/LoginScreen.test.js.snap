// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in component <HomeScreen/> Should show the component properly 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
